packages/node-cli/src/utils/ApplyInstallationPreferencesUtils.js

```diff
@@ -0,0 +1,60 @@
+/*
+ ** Copyright (c) 2020 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+
+'use strict';
+
+const NodeTranslationService = require('../services/NodeTranslationService');
+const ProjectInfoService = require('../services/ProjectInfoService');
+const { LINKS, PROJECT_ACP } = require('../ApplicationConstants');
+const NodeConsoleLogger = require('../loggers/NodeConsoleLogger');
+
+const APPLY_CONTENT_PROTECTION = 'applycontentprotection';
+const APPLY_INSTALLATION_PREFERENCES = 'applyinstallprefs';
+
+const {
+	UTILS: {
+		APPLY_CONTENT_PROTECTION_ARGUMENT_HANDLER: { ERRORS, WARNINGS },
+		APPLY_INSTALLATION_PREFERENCES_ARGUMENT_HANDLER: { INSTALLATION_PREFERENCES_ERRORS },
+	},
+} = require('../services/TranslationKeys');
+
+function validate(args, projectFolder, commandName) {
+	const projectInfoService = new ProjectInfoService(projectFolder);
+
+	if (args[APPLY_CONTENT_PROTECTION]) {
+		NodeConsoleLogger.warning(NodeTranslationService.getMessage(WARNINGS.APPLY_CONTENT_PROTECTION_IS_DEPRECATED));
```

Comment:
We should not use directly the NodeConsoleLogger. Instead pass it from the commandAction that is calling  `ApplyInstallationPreferencesUtils.validate()`, the logger will properly injected to the specific commandAction from the BaseAction to its `this._log` property.